lib/ansible/modules/network/fortios/fortios_address_group.py

```diff
@@ -0,0 +1,236 @@
+extends_documentation_fragment: fortios
+options:
+  name:
+    description:
+      - Name of the address group.
+    required: true
+    aliases:
+      - group_name
+  member:
+    description:
+      - Member(s) address name to add or delete.
+    required: true
+    aliases:
+      - members
+  state:
+    description:
+      - Specifies if address need to be added or deleted in group.
+    default: 'present'
+    choices: ['present', 'absent']
+  comment:
+    description:
+      - free text to describe address group.
+notes:
+  - This module requires pyFG python library.
+"""
+
+EXAMPLES = """
+- name: Add google DNS address object in dns_servers address group
+  fortios_address_group:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: present
+    name: dns_servers
+    member: google_dns
+    comment: "Public DNS servers"
+
+- name: Remove yahoo from trusted mail providers
+  fortios_address_group:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: absent
+    name: trusted_mail_providers
+    member: yahoo_server
+    comment: "Trusted Email providers"
+"""
+
+RETURN = """
+firewall_address_config:
+  description: full firewall adresses config string
+  returned: always
+  type: string
+change_string:
+  description: The commands executed by the module
+  returned: only if config changed
+  type: string
+"""
+
+import re
+import shlex
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+from ansible.module_utils.fortios import fortios_argument_spec, fortios_required_if
+from ansible.module_utils.fortios import backup
+
+#check for pyFG lib
+try:
+    from pyFG import FortiOS, FortiConfig
+    from pyFG.exceptions import CommandExecutionException, FailedCommit
+    HAS_PYFG=True
+except:
+    HAS_PYFG=False
+
+
+def is_invalid_name(input_str):
+    #char must be letters, digits, - , _ , .
+    #must have between 1 and 63 chars
+    reg=re.compile(r'^[a-zA-Z0-9\.\-_]{1,63}$')
+    return not reg.match(input_str)
+
+def main():
+    #define module params
+    argument_spec = dict(
+        state       = dict(choices=['present', 'absent'], default='present'),
+        name        = dict(aliases=['group_name'], required=True, type='str'),
+        member      = dict(aliases=['members'], required=True, type='list'),
+        comment     = dict(type='str'),
+    )
+
+    #merge global argument_spec from module_utils/fortios.py
+    argument_spec.update(fortios_argument_spec)
+
+    #decalre module
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=fortios_required_if,
+    )
+
+    #check params
+    if is_invalid_name(module.params['name']):
+        module.fail_json(msg="Bad name argument value %s, must contain only letters, digits, -, _, ." % (module.params['name']))
+
+    for member in module.params['member']:
```

Comment:
You could append the `name` parameter here (using something like `in module.params['member'] + [module.params['name']]`).